### PR TITLE
DOC: improve explanation of con argument DataFrame.to_sql

### LINKED ISSUE #34824: DOC: to_sql accepting Engine vs. Connectable
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_sql.html

#### Documentation problem

[Documentation](https://github.com/pandas-dev/pandas/blob/1ce1c3c1ef9894bf1ba79805f37514291f52a9da/pandas/core/generic.py#L2530) says that `con` is to be a "sqlalchemy.engine.Engine or ..." while [source](https://github.com/pandas-dev/pandas/blob/f34fd588cd9facfa82a792fb570fb8a56f11f6c7/pandas/io/sql.py#L539) says that `con` can be a "SQLAlchemy connectable(engine/connection) or ..."

#### Suggested fix for documentation

Documentation should mention that a SQLAlchemy `Connection` is acceptable because some operations like [an upsert via a temporary table in SQL Server](https://stackoverflow.com/q/62388767/2144390) need to ensure that the same DBAPI connection is used for the entire operation. (SQL Server local temp tables are only visible to the connection/session that created them).

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #34824 
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2576,18 +2576,27 @@ def to_sql(
         >>> engine.execute("SELECT * FROM users").fetchall()
         [(0, 'User 1'), (1, 'User 2'), (2, 'User 3')]
 
-        >>> df1 = pd.DataFrame({'name' : ['User 4', 'User 5']})
-        >>> df1.to_sql('users', con=engine, if_exists='append')
+        An `sqlalchemy.engine.Connection` can also be passed to to `con`:
+        >>> with engine.connect() as connection:
```
Comment (gordthompson) [inline pandas/core/generic.py:2580]: `with engine.begin() as connection:` would be a safer bet. Upcoming changes in SQLAlchemy will (probably) result in both `engine.connect()` and `engine.begin()` returning a `Connection` with an active transaction, but the default behaviour on exit from the context manager block will (probably) be that for `.connect()` the transaction will be rolled back, while with `.begin()` the transaction will be committed.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2576,18 +2576,27 @@ def to_sql(
...
+        ...     df1 = pd.DataFrame({'name' : ['User 4', 'User 5']})
+        ...     df1.to_sql('users', con=connection, if_exists='append')
+
+        This is allowed to support operations that require that the same
+        DBAPI connection is used to the entire operation.
```
Comment (gordthompson) [inline pandas/core/generic.py:2585]: ... used for the entire ...

**Review by gordthompson — CHANGES_REQUESTED** [review]
Looks great overall. Just a couple of minor suggestions.

File: pandas/core/generic.py (hunk 1 above)
Comment (arw2019) [reply pandas/core/generic.py:2580]: Sounds good - fixed it

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @arw2019

**Pull request merged**